tests/dialects/test_x86.py

```diff
@@ -27,3 +29,59 @@
 def test_register(register: x86.register.GeneralRegisterType, name: str):
     assert register.is_allocated
     assert register.register_name == name
+
+
+def test_push_op():
```

Comment:
I would recommend sticking to filecheck tests for everything that can be checked with filecheck, which I think is the case for these. For verification errors search for `--verify-diagnostics` in the project to find some examples. Having the expected error string is quite useful.